Refactor/bc 217 test migrate header

mendo94 opened the pull request:
Please also review changes made in the other files: SingleActionButton.scss, Landing.jsx, SignIn.jsx. The styling from the Single Action Button is not being applied if it is wrapped in the form tag.

mendo94 committed:
Error coming from SAB stylesheet, floating code

mendo94 committed:
Updated bc SAB is integrated, style is not applied

mendo94 committed:
change navigation to correct file paths

mendo94 committed:
Added payload prop to signup's SAB to remove error

mendo94 committed:
there is not a need for header text, removed

mendo94 committed:
Changed heading tag to paragraph tag

mendo94 committed:
double check SAB scss file

mendo94 committed:
added js documentation

yeezick commented on client/src/components/Header/Header.jsx:
```diff
@@ -1,11 +1,16 @@
-  </header>
-  )
+/**
+ *
```
extra line, but love that you used JSDocs!!

yeezick commented on client/src/screens/SignUp/SignUp.jsx:
```diff
@@ -26,6 +26,11 @@ export const SignUp = () => {
 
+  const routeToCreateProfile = {
+    handler: null,
+    title: 'Registration',
```
title should be "register"

yeezick commented on client/src/screens/SignUp/SignUp.jsx:
```diff
@@ -157,7 +162,7 @@ export const SignUp = () => {
           )}
         </div>
-        <SingleActionButton text="Register" type="submit" />
+        <SingleActionButton payload={routeToCreateProfile} text="Register" type="submit" />
```
can remove text & type props since SAB only takes style & size now

yeezick commented on client/src/components/Form/Form.jsx:
```diff
@@ -36,7 +40,7 @@ export const Form = ({ formData, formState }) => {
         </div>
       ))}
-      <SingleActionButton text={button.text} type="submit" />
+      <SingleActionButton payload={submitFormPayload} text={button.text} type="submit" />
```
text & type can be removed but title should be passed in as payload

yeezick committed:
add title to form component SAB payload
remove unused props from SAB instances
delete form class conflicting with SAB styles
delete unnecessary conflicting classes => auth-form from register/log in screens
import Header into SignUp
clean up unresolved conversations from PR

yeezick reviewed (approved):
Good work @mendo94! The conversations I posted above were easy enough for me to give a quick fix. 
The problem with the form buttons was that there was a selector that targeted buttons within that form, so the form selector was overwriting the SAB classes.
